app/api/ada/lib/storage/database/README.md

```diff
@@ -0,0 +1,96 @@
+# Architecture
+
+This folder handles interfacing with [lovefield](https://github.com/google/lovefield) which is a library that allows to write SQL-like queries in pure Javascript.
+
+Folders are split based on the categories inside the [storage spec](../../../../../../docs/specs/code/STORAGE.md)
+
+The main requirement of this architecture is that it must work well with [transactional queries](https://en.wikipedia.org/wiki/Database_transaction). Transactions in Lovefield work by requiring you to lock each table that will be used in the query BEFORE the query executes.
+
+Notably:
+
+1) If you query a table without locking it first, it is a runtime error
+2) If you query a table that is already locked, you will block until it gets unlocked (note: risk of deadlock)
+
+To avoid bugs, every high-level query is wrapped in a class with the following architecture
+
+```js
+export class MyTopLevelQuery {
+  /** ONLY tables that I access directly **/
+  static ownTables = Object.freeze({
+    [Tables.SomeTableYouNeed.name]: Tables.SomeTableYouNeedSchema,
+  });
+
+  /** other high level queries you use inside this query **/
+  static depTables = Object.freeze({
```

Comment:
not a big thing, but it wasn’t obvious for me that `dep` stand for `dependencies`